app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -796,7 +802,15 @@ class WebViewActivity : BaseActivity(), io.homeassistant.companion.android.webvi
                 startActivity(SettingsActivity.newInstance(this))
             }
             alert.setNegativeButton(R.string.refresh) { _, _ ->
-                webView.reload()
+                runBlocking {
+                    failedConnection = if (failedConnection == "external") {
+                        webView.loadUrl(urlRepository.getUrl(true).toString())
+                        "internal"
+                    } else {
+                        webView.loadUrl(urlRepository.getUrl(false).toString())
+                        "external"
+                    }
+                }
```

Comment:
Rather than trying both urls we should make sure we are just trying to load the correct url instead.  You should just have to call `webView.loadUrl(urlRepository.getUrl().toString())` I believe.

Reply:
But the way this is currently implemented there is no way for a user to know which URL is going to be used.

Reply:
We can post a toast on the device saying which URL is going to be tried or I can update the text in the pop-up to let them know which one failed.  I think offering another button may confuse/complicate things since not everyone uses internal connection URL.

Reply:
Looks like the alert dialog does not allow to use variables in `setMessage` I think the current message might suffice since it tells users hitting refresh cycles through the URLs.  If we think a toast message would be good I can add that too, we already know the user is looking at the device :)

Reply:
ok we will now update the refresh button to reflect which URL will be attempted next :) the pop-up message is now updated to reflect this too.

![image](https://user-images.githubusercontent.com/1634145/107100342-eea76500-67c8-11eb-8e98-8b460311f4fe.png)

![image](https://user-images.githubusercontent.com/1634145/107100357-f6670980-67c8-11eb-878b-765b4aaa3083.png)

Reply:
fixed the label logic so users who don't use internal URL will not see "refresh internal url" :)